Source/It.cs

```diff
@@ -55,15 +55,15 @@ public static class It
 		public static TValue IsAny<TValue>()
 		{
 			return Match<TValue>.Create(
-				value => value == null || typeof(TValue).IsAssignableFrom(value.GetType()),
+				value => value == null || value is TValue,
 				() => It.IsAny<TValue>());
 		}
 
 		/// <include file='It.xdoc' path='docs/doc[@for="It.IsNotNull"]/*'/>
 		public static TValue IsNotNull<TValue>()
 		{
 			return Match<TValue>.Create(
-				value => value != null && typeof(TValue).IsAssignableFrom(value.GetType()),
+				value => value is TValue,
```

Comment:
The null check can be incorporated into the type check, since `null is …` is always false.